### PR TITLE
[Refactor] Refactor event based compaction framework

### LINKED ISSUE #10503: Refactor compaction policy support frequency ingestion

### PR DESCRIPTION
**Pull request by meegoo**
## What type of PR is this：
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [x] Refactor
- [ ] UT
- [ ] Doc
- [ ] Tool

## Which issues of this PR fixes ：

Fixes #10503

## Problem Summary(Required) ：

This PR only contain refactor of `event base` compaction framework
1. optimize CompactionManager's update_tablet(), avoid multiple calls and fully copy
2. support compaction on missing version tablet
3. optimize default base compaction score calculation strategy
4. optimize compaction scheduler avoid create unnecessary compaction task
5. optimize cumulative compaction scheduler interval
6. fix duplicate compaction bug

the `default compaction strategy` still need optimize after some test, meanwhile we will support `size-tiered` compaction strategy in subsequent PRs.

## Checklist:

- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

### CODE REVIEW

**Comment by dulong41** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_be_unittest

**Comment by dulong41** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

File: be/src/storage/compaction_manager.cpp [hunk 1]
```diff
@@ -45,58 +49,161 @@ void CompactionManager::update_candidates(std::vector<CompactionCandidate> candi
...
 }
 
-void CompactionManager::update_tablet_async(const TabletSharedPtr& tablet, bool need_update_context,
-                                            bool is_compaction) {
-    Status st = _update_candidate_pool->submit_func([tablet, need_update_context, is_compaction, this] {
-        update_tablet(tablet, need_update_context, is_compaction);
-    });
-    if (!st.ok()) {
-        LOG(WARNING) << "update candidate failed. status:" << st.to_string();
+void CompactionManager::update_tablet_async(TabletSharedPtr tablet) {
```
Comment (chaoyli) [inline be/src/storage/compaction_manager.cpp:181]: Why not call it when add_rowset in tablet. It's the only entrance.

File: be/src/storage/compaction_manager.h [hunk 2]
```diff
@@ -38,12 +38,11 @@ class CompactionManager {
 
     void update_candidates(std::vector<CompactionCandidate> candidates);
```
Comment (chaoyli) [inline be/src/storage/compaction_manager.h:40]: I think the CompactionManager and CompactionScheduler may have some overlaps.

File: be/src/storage/CMakeLists.txt [hunk 3]
```diff
@@ -149,15 +149,14 @@ add_library(Storage STATIC
...
-    compaction_context.cpp
     compaction_task.cpp
     compaction_utils.cpp
     compaction_manager.cpp
     compaction_scheduler.cpp
     horizontal_compaction_task.cpp
     vertical_compaction_task.cpp
     compaction_task_factory.cpp
-    base_and_cumulative_compaction_policy.cpp
+    default_cumulative_compaction_policy.cpp
```
Comment (chaoyli) [inline be/src/storage/CMakeLists.txt:159]: Take a new name?

File: be/src/storage/compaction_manager.cpp (hunk 1 above)
Comment (meegoo) [reply be/src/storage/compaction_manager.cpp:181]: not the only entrance, suck as clone not use that

File: be/src/storage/compaction_manager.h (hunk 2 above)
Comment (meegoo) [reply be/src/storage/compaction_manager.h:40]: it can be unify to one class, but if we want to support multi scheduler, it may be useful later.

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
run starrocks_be_unittest

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Review by chaoyli — APPROVED** [review]

**Review by Astralidea — APPROVED** [review]

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
@mergify backport branch-2.5

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by wanpengfei-git** [comment]
run starrocks_admit_test

**Comment by meegoo** [comment]
@mergify backport branch-2.5

**Pull request merged**